### PR TITLE
cstor(pool_chaos): check the app status after bringing down pool

### PR DESCRIPTION
**Pull request by vishnuitta**
This PR is to add a litmus job that does the following:
- checks that app is running fine
- verifies for provisioning by looking for cvr and their status
- bring down the cstor pool pod
- checks that app is running fine still
- bring down cstor pool pod again (can be same or different one)
- checks that app is running fine still

This also takes care of ReplicaCount provided in StorageClass.

This doesn't include setting the liveness pod related environment variables, which need to be done along with bash script for e2e.

Signed-off-by: Vishnu Itta <vitta@mayadata.io>

### CODE REVIEW

File: apps/percona/chaos/cstor_pool_failure/run_litmus_test.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                fieldPath: metadata.name
+          - name: MY_POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          # spec.volumes is not supported via downward API
+          - name: MY_POD_HOSTPATH
+            value: /mnt/chaos/cstor_pool_failure
+        command: ["/bin/bash"]
+        args: ["-c", "./logger.sh -d ansibletest -r maya,openebs,pvc,percona; exit 0"]
```
Comment (gprasath) [inline apps/percona/chaos/cstor_pool_failure/run_litmus_test.yml]: @vishnuitta , shall we include 'cstor' string here to catch the logs from those pods too.

File: chaoslib/openebs/cstor_pool_failure.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  register: sc
+
+- name: Derive ReplicaCount from SC
+  shell: >
+    kubectl get sc {{ sc.stdout }} -n {{ operator_ns }} -o json |
+    jq '.metadata.annotations."cas.openebs.io/config"' -r |
+    grep -A 1 -w "name: ReplicaCount" | grep -w value | awk '{print $2}'
+  args:
+    executable: /bin/bash
+  register: replicacount
```
Comment (gprasath) [inline chaoslib/openebs/cstor_pool_failure.yaml]: Shall we derive this using deployment.

Comment (vishnuitta) [reply chaoslib/openebs/cstor_pool_failure.yaml]: deployment will not have replicacount

File: apps/percona/chaos/cstor_pool_failure/run_litmus_test.yml (hunk 1 above)
Comment (vishnuitta) [reply apps/percona/chaos/cstor_pool_failure/run_litmus_test.yml]: yes @gprasath .. it should be good.. will add that

File: apps/percona/chaos/cstor_pool_failure/test.yml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            app: ""
+            phase: completed
+            verdict: "{{ flag }}"
+           
+        - name: Apply the litmus result CR
+          shell: kubectl apply -f litmus-result.yaml
+          args:
+            executable: /bin/bash
+          register: lr_status 
+          failed_when: "lr_status.rc != 0"
```
Comment (ksatchit) [inline apps/percona/chaos/cstor_pool_failure/test.yml]: There is a util we can include to generate and update the result CR now : https://github.com/openebs/litmus/blob/master/common/utils/update_litmus_result_resource.yml

File: chaoslib/openebs/cstor_pool_failure.yaml (hunk 2 above)
Comment (ksatchit) [reply chaoslib/openebs/cstor_pool_failure.yaml]: The grep becomes a necessity here as we hit a dead-end upon extracting the content (params) from the cas.openebs.io/config annotation !

File: chaoslib/openebs/cstor_pool_failure.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  register: cvr_count
+
+- name: Compare ReplicaCount and cvr_count to verify provisioning
+  debug:
+    msg:
+      - "replicaCount: {{ replicacount.stdout }}"
+      - "replicaCountInt: {{ replicacount.stdout | int }}"
+      - "replicacnt: {{ replicacnt }}"
+      - "cvr_count: {{ cvr_count.stdout | int }}"
+  failed_when: "replicacnt|int != cvr_count.stdout|int"
```
Comment (ksatchit) [inline chaoslib/openebs/cstor_pool_failure.yaml]: :+1:

File: chaoslib/openebs/cstor_pool_failure.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
+- name: Derive PV from application PVC
```
Comment (ksatchit) [inline chaoslib/openebs/cstor_pool_failure.yaml]: This is a good util to check volume status/sanity before proceeding to the chaos (pool deployment delete). What do you think about renaming this to cstor_volume_sanity.yaml - we could include this as a pre-chaos step in every cstor test

Comment (ksatchit) [reply chaoslib/openebs/cstor_pool_failure.yaml]: The cstor pool delete could then be called separately from the test? i.e., the chaosutil selection jinja template could call this instead.

File: chaoslib/openebs/cstor_pool_failure.yaml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+- name: Check status of cvr
+  command: echo "{{ item }}"
+  failed_when: "item != \"Online\""
+  with_items:
+    - "{{ cvr_status_phase.stdout_lines }}"
+
+- include: cstor_delete_pool_deployment.yaml
+
+- include: cstor_delete_pool_deployment.yaml
```
Comment (ksatchit) [inline chaoslib/openebs/cstor_pool_failure.yaml]: A good test to do it more than once, as shuf will ensure selection of a different cvr (and thereby pool) most times. Would it be good to define the chaos iteration count in the test_vars & also include a volume sanity check before each iteration of chaos - so we will fail faster, if at all.

File: tools/ansible-runner/Dockerfile [hunk 7]
```diff
@@ -6,7 +6,7 @@ LABEL maintainer="OpenEBS"
 RUN rm -rf /var/lib/apt/lists/* && \
     apt-get clean && \
     apt-get update --fix-missing || true && \
-    apt-get install -y python-minimal python-pip \
+    apt-get install -y jq python-minimal python-pip \
```
Comment (ksatchit) [inline tools/ansible-runner/Dockerfile]: We could go ahead with this - just wanted to see if we can use kubectl natively (there are some thoughts around bundling lib/tests so they can downloaded/runnable in/outside containers - i.e., wherever kubeconfig is present)

**Comment by ksatchit** [comment]
Thanks @vishnuitta , this is very helpful! Suggested some minor changes, which can be discussed.

File: apps/percona/chaos/cstor_pool_failure/test.yml (hunk 3 above)
Comment (vishnuitta) [reply apps/percona/chaos/cstor_pool_failure/test.yml]: done

File: chaoslib/openebs/cstor_pool_failure.yaml (hunk 5 above)
Comment (vishnuitta) [reply chaoslib/openebs/cstor_pool_failure.yaml]: I made this as cstor_verify_pool_provisioning.yml, and, included this in cstor_pool_failure.yml.
This file cstor_verify_pool_provisioning.yml can be moved to different place to use this from other directories.

File: chaoslib/openebs/cstor_pool_failure.yaml (hunk 6 above)
Comment (vishnuitta) [reply chaoslib/openebs/cstor_pool_failure.yaml]: shuf can give the same cvr also, but, that should be fine for us even if it is different.
This same file verifies that deployment is started again.

File: tools/ansible-runner/Dockerfile (hunk 7 above)
Comment (vishnuitta) [reply tools/ansible-runner/Dockerfile]: removed the dependency on jq

**Pull request merged**